app/src/test/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModelTest.kt

```diff
@@ -152,6 +152,40 @@ class NewConversationViewModelTest {
         viewModel.groupOptionsState.error.shouldBeNull()
     }
 
+    @Test
+    fun `given create group conflicted backends error, when clicked discard group, then error should be cleaned`() =
+        runTest {
+            val (_, viewModel) = NewConversationViewModelArrangement()
+                .withIsSelfTeamMember(true)
+                .withConflictingBackendsFailureOnCreatingGroup()
+                .arrange()
+
+            viewModel.createGroup()
+            advanceUntilIdle()
+            assert(viewModel.groupOptionsState.error?.isConflictedBackends ?: false)
+
+            viewModel.onDiscardGroupCreationClick()
+            advanceUntilIdle()
+            viewModel.groupOptionsState.error.shouldBeNull()
+        }
```

Comment:
We are testing two different things in one test here.
Let's just focus on `onDiscardGroupCreationClick` scope, we don't need to call createGroup() and assert the creation.
We should go simplier:
```
   @Test
    fun `given failure to create group, when user clicked on discard group, then navigate to Home`() =
        runTest {
            val (_, viewModel) = NewConversationViewModelArrangement()
                .withConflictingBackendsFailure()
                .arrange()

            viewModel.onDiscardGroupCreationClick()

            advanceUntilIdle()
            viewModel.groupOptionsState.error.shouldBeNull()
             coVerify(exactly = 1) { navigationManager.navigate(NavigationCommand(NavigationItem.Home.getRouteWithArgs(), BackStackMode.CLEAR_WHOLE)) }

        }
```

Reply:
I don't agree because we cannot init error state without triggering createGroup() But I will add this
`             coVerify(exactly = 1) { navigationManager.navigate(NavigationCommand(NavigationItem.Home.getRouteWithArgs(), BackStackMode.CLEAR_WHOLE)) }
`

Reply:
You can add a function in NewConversationViewModelArrangement to update the state to ConflictedBackends in viewModel
```
fun withConflictingBackendsFailure() {
        viewModel.groupOptionsState = viewModel.groupOptionsState.copy(
            error = GroupOptionState.Error.ConflictedBackends(listOf())
        )
    }
    ```